### PR TITLE
Add a /report.json endpoint to receive a complete bundleSizes JSON object

### PR DESCRIPTION
**Pull request by danielrozenberg**
Related to #271

### CODE REVIEW

File: bundle-size/app.js [hunk 1]
```diff
@@ -564,6 +564,60 @@ module.exports = app => {
     }
   });
 
+  v0.post('/commit/:headSha/report.json', async (request, response) => {
+    const {headSha} = request.params;
+    const {baseSha, bundleSizes} = request.body;
+
+    if (typeof baseSha !== 'string' || !/^[0-9a-f]{40}$/.test(baseSha)) {
```
Comment (rsimha) [inline bundle-size/app.js]: Would it make sense to extract this into a `isValidSha` function?

File: bundle-size/app.js [hunk 2]
```diff
@@ -564,6 +564,60 @@ module.exports = app => {
...
+
+    // TODO(#271, danielrozenberg): this is a transitionary solution. This API
+    // endpoint accepts a JSON with multiple bundle sizes, but for now it only
+    // processes the size of dist/v0.js. Once we determine how we want to report
+    // and block PRs based on other bundle sizes we can start processing them.
+    const bundleSize = bundleSizes['dist/v0.js'];
+
+    const check = await getCheckFromDatabase(headSha);
+    if (!check) {
+      return response.status(404).end(`${headSha} not in database`);
```
Comment (rcebulko) [inline bundle-size/app.js]: Nit: A bit more detail in the error message (what database) might be helpful

Comment (rsimha) [inline bundle-size/app.js]: If this message appears in the status check or  Travis logs, perhaps expand on what database is being referred to?

File: bundle-size/test/app.test.js [hunk 3]
```diff
@@ -849,6 +852,491 @@ describe('bundle-size', () => {
...
 
+    describe('/commit/:headSha/report.json', () => {
+      let jsonPayload;
+
+      beforeEach(() => {
+        jsonPayload = {
+          baseSha: '5f27002526a808c5c1ad5d0f1ab1cec471af0a33',
+          bundleSizes: {
+            'dist/v0.js': 12.34,
+            'dist/amp4ads-v0.js': 11.22,
```
Comment (rsimha) [inline bundle-size/test/app.test.js]: Yay! (I know it’s just a test, but still.)

/cc @lannka

File: bundle-size/app.js (hunk 1 above)
Comment (rcebulko) [reply bundle-size/app.js]: I've thought the same thing a few times, I feel like I've seen this in a handful of places in this app (though idr how many were the same N occurrences). Might be worth `isValidSha` or `assertValidSha` that throws

File: bundle-size/app.js (hunk 2 above)
Comment (danielrozenberg) [reply bundle-size/app.js]: Done

File: bundle-size/app.js (hunk 1 above)
Comment (danielrozenberg) [reply bundle-size/app.js]: Not really, this is only used once, and once again for a soon-to-be-deleted endpoint. I'd rather not extract this into a method in that case :)

File: bundle-size/app.js (hunk 2 above)
Comment (danielrozenberg) [reply bundle-size/app.js]: Done

**Pull request merged**